cmd/thanos/store.go

```diff
@@ -182,6 +183,9 @@ func (sc *storeConfig) registerFlag(cmd extkingpin.FlagClause) {
 	cmd.Flag("store.index-header-lazy-reader-idle-timeout", "If index-header lazy reader is enabled and this idle timeout setting is > 0, memory map-ed index-headers will be automatically released after 'idle timeout' inactivity.").
 		Hidden().Default("5m").DurationVar(&sc.lazyIndexReaderIdleTimeout)
 
+	cmd.Flag("store.enable-lazy-expanded-postings", "If true, Store Gateway will estimate postings size and try to lazily expand postings if it downloads less data than expanding all postings.").
```

Comment:
Why not just enable it by default? Your code will get feedback faster. You could still keep this option but I'd suggest enabling it by default.

Reply:
I still prefer to make it disabled by default as it is experimental feature to me and most of the usecase with not that large scale, this flag doesn't help much. CX can enable it if needed.

Another reason is that this pr is already pretty large... I don't want to make it default and then need to change some existing E2E tests :)

Reply:
Makes sense :+1: